embedded_rtic/examples/sonars.rs

```diff
@@ -0,0 +1,440 @@
+            if gpiote.channel0().is_event_triggered() {
+                info!("Interrupt on channel 0, {:?}", debug_time);
+                triggered_sonars[0] = Some(SonarChannel::Forward);
+            }
+            if gpiote.channel1().is_event_triggered() {
+                info!("Interrupt on channel 1, {:?}", debug_time);
+                triggered_sonars[1] = Some(SonarChannel::Left);
+            }
+            if gpiote.channel2().is_event_triggered() {
+                info!("Interrupt on channel 2, {:?}",debug_time);
+                triggered_sonars[2] = Some(SonarChannel::Right);
+            }
+        });
+
+        // Print messages for each triggered sonar
+        for sonar in triggered_sonars.iter().filter_map(|x| *x) {
+            match sonar {
+                SonarChannel::Forward => {
+                    //info!("Event triggered by Sonar 1 (Forward)");
+                    if cx.local.times[0] == zero {
+                        cx.local.times[0] = time;
+                    } else {
+                        let distance = time.checked_duration_since(cx.local.times[0]).unwrap().to_micros() / 56;
+                        match cx.local.sender.try_send(distance as u32) {
+                            Ok(_) => {}
+                            _ => {
+                                error!("Distance FORWARD did not fit in to the buffer, scheduling is probably broken.");
+                                // This is not fatal so we simply continue with our life
+                            }
+                        }
+                        cx.local.times[0] = zero;
+                    }
+                }
+                SonarChannel::Left => {
+                    //info!("Event triggered by Sonar 2 (Left)");
+                    if cx.local.times[1] == zero {
+                        cx.local.times[1] = time;
+                    } else {
+                        let distance2 = time.checked_duration_since(cx.local.times[1]).unwrap().to_micros() / 56;
+                        cx.local.times[1] = zero;
+                        match cx.local.sender2.try_send(distance2 as u32) {
+                            Ok(_) => {}
+                            _ => {
+                                error!("Distance LEFT did not fit in to the buffer, scheduling is probably broken.");
+                                // This is not fatal so we simply continue with our life
+                            }
+                        }
+                    }
+                }
+                SonarChannel::Right => {
+                    //info!("Event triggered by Sonar 3 (Right)");
+                    if cx.local.times[2] == zero {
+                        cx.local.times[2] = time;
+                    } else {
+                        let distance3 = time.checked_duration_since(cx.local.times[2]).unwrap().to_micros() / 56;
+                        info!("distance 3 before sending: {:?}", distance3);
```

Comment:
change these to `trace!` since they are mainly useful as a debug trace if you really want to see what is happening.